### PR TITLE
Machine code fix + missing relocations

### PR DESCRIPTION
**Pull request by lakor64**
- Fixed bad machine code for Alpha 32-bit and, MIPS R3000 and MIPS R10000
- Added Hitachi SH3, Loongarch and PPC BE machines
- Added remaining base relocations

Info gathered from MSDN PE Format documentation page and radare2 pe header file

### CODE REVIEW

File: pe-parser-library/include/pe-parse/nt-headers.h [hunk 1]
```diff
@@ -60,20 +60,24 @@ constexpr std::uint16_t IMAGE_FILE_MACHINE_CEF = 0xcef;
...
 constexpr std::uint16_t IMAGE_FILE_MACHINE_MIPSFPU = 0x366;   // MIPS with FPU
 constexpr std::uint16_t IMAGE_FILE_MACHINE_MIPSFPU16 = 0x466; // MIPS16 with FPU
 constexpr std::uint16_t IMAGE_FILE_MACHINE_POWERPC = 0x1f0;   // Power PC little endian
 constexpr std::uint16_t IMAGE_FILE_MACHINE_POWERPCFP = 0x1f1; // Power PC with floating point support
-constexpr std::uint16_t IMAGE_FILE_MACHINE_R3000 = 0x166;     // MIPS little endian, 0x160 big-endian
+constexpr std::uint16_t IMAGE_FILE_MACHINE_POWERPCBE = 0x1f2; // Power PC big endian
+constexpr std::uint16_t IMAGE_FILE_MACHINE_R3000 = 0x162;     // MIPS little endian, 0x160 big-endian
 constexpr std::uint16_t IMAGE_FILE_MACHINE_R4000 = 0x166;     // MIPS little endian
-constexpr std::uint16_t IMAGE_FILE_MACHINE_R10000 = 0x166;    // MIPS little endian
+constexpr std::uint16_t IMAGE_FILE_MACHINE_R10000 = 0x168;    // MIPS little endian
```
Comment (woodruffw) [inline pe-parser-library/include/pe-parse/nt-headers.h:74]: This came from Radare, right? I don't see it in the MSDN docs.

File: pe-parser-library/include/pe-parse/nt-headers.h [hunk 2]
```diff
@@ -60,20 +60,24 @@ constexpr std::uint16_t IMAGE_FILE_MACHINE_CEF = 0xcef;
...
 constexpr std::uint16_t IMAGE_FILE_MACHINE_RISCV32 = 0x5032;  // RISC-V 32-bit address space
 constexpr std::uint16_t IMAGE_FILE_MACHINE_RISCV64 = 0x5064;  // RISC-V 64-bit address space
 constexpr std::uint16_t IMAGE_FILE_MACHINE_RISCV128 = 0x5128; // RISC-V 128-bit address space
 constexpr std::uint16_t IMAGE_FILE_MACHINE_SH3 = 0x1a2;       // Hitachi SH3
 constexpr std::uint16_t IMAGE_FILE_MACHINE_SH3DSP = 0x1a3;    // Hitachi SH3 DSP
+constexpr std::uint16_t IMAGE_FILE_MACHINE_SH3E = 0x1a4;      // Hitachi SH3E
```
Comment (woodruffw) [inline pe-parser-library/include/pe-parse/nt-headers.h:80]: Same question for this one 🙂

**Review by woodruffw — APPROVED** [review]
LGTM! A few questions about where the constants came from.

**Comment by woodruffw** [comment]
Lint failing, FYI: https://github.com/trailofbits/pe-parse/actions/runs/6283000274/job/17103908037?pr=193

**Comment by woodruffw** [comment]
Got it, thanks! The lint needs to be fixed, but otherwise this LGTM.

**Comment by lakor64** [comment]
> Got it, thanks! The lint needs to be fixed, but otherwise this LGTM.

Is it related to formatting issue I suppose? I have to run clang-format?

**Comment by woodruffw** [comment]
> Is it related to formatting issue I suppose? I have to run clang-format?

Yep, correct. There should be a target for it in the CMake-based build.

**Comment by lakor64** [comment]
Should be ok now

**Review by woodruffw — APPROVED** [review]

**Comment by woodruffw** [comment]
Thanks @lakor64!

**Pull request merged**